grammars/css.cson

```diff
@@ -20,294 +20,1218 @@
+        'beginCaptures':
+          '1':
+            'name': 'keyword.control.at-rule.import.css'
+          '2':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '\\s*(?=;|})'
+        'name': 'meta.at-rule.import.css'
+        'patterns': [
+          {
+            'begin': '\\G\\s*(?=/\\*)'
+            'end': '(?<=\\*/)\\s*'
+            'patterns': [
+              {
+                'include': '#comment-block'
+              }
+            ]
+          }
+          {
+            'include': '#string-double'
+          }
+          {
+            'include': '#string-single'
+          }
+          {
+            'include': '#comment-block'
+          }
+          {
+            'include': '#url'
+          }
+          {
+            'include': '#media-query-list'
+          }
+        ]
+      }
+      {
+        # @font-face
+        'begin': '(?i)((@)font-face)\\s*(?={|$|/\\*)'
+        'beginCaptures':
+          '1':
+            'name': 'keyword.control.at-rule.font-face.css'
+          '2':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '(?!\\G)'
+        'name': 'meta.at-rule.font-face.css'
+        'patterns': [
+          {
+            'include': '#comment-block'
+          }
+          {
+            'include': '#escapes'
+          }
+          {
+            'include': '#rule-list'
+          }
+        ]
+      }
+      {
+        # @media
+        'begin': '(?i)(?=@media(?:\\s|\\(|$|/\\*))'
+        'end': '\\s*(})|(?=;)'
+        'endCaptures':
+          '1':
+            'name': 'punctuation.section.property-list.end.css'
+        'patterns': [
+          {
+            'begin': '(?i)(@)media(?=\\s|\\(|$|/\\*)'
+            'beginCaptures':
+              '0':
+                'name': 'keyword.control.at-rule.media.css'
+              '1':
+                'name': 'punctuation.definition.keyword.css'
+            'end': '\\s*(?=[;{])'
+            'name': 'meta.at-rule.media.css'
+            'patterns': [
+              {
+                'include': '#media-query-list'
+              }
+            ]
+          }
+          {
+            'begin': '\\s*({)'
+            'beginCaptures':
+              '1':
+                'name': 'punctuation.section.property-list.begin.css'
+            'end': '(?=})'
+            'patterns': [
+              {
+                'include': '$self'
+              }
+            ]
+          }
+        ]
+      }
+      {
+        # @page
+        'begin': '(?i)(@)page(?=[\\s:{;]|$|/\\*)'
+        'captures':
+          '0':
+            'name': 'keyword.control.at-rule.page.css'
+          '1':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '\\s*(?=$|[{:;])'
+        'name': 'meta.at-rule.page.css'
+        'patterns': [
+          {
+            'include': '#rule-list'
+          }
+        ]
+      }
+      {
+        # @counter-style
+        'begin': '(?i)((@)counter-style)(?=[\\s{\'";]|/\\*|$)\\s*'
```

Comment:
For rules which take an identifier before the block, it's a way of trimming whitespace from the segment that's scoped by `contentName`. It might make a difference for themes which apply background colours to `variable.name` or whatever:

<img src="https://cloud.githubusercontent.com/assets/2346707/19719602/649745a2-9bb6-11e6-935d-16b8eb968f70.gif" width="216" alt="Figure 1" />

However, not every rule is affected by this; in fact, I only added it in the first place so I wouldn't have to worry about precedence issues. I suppose now that all the specs have been added and everything else fleshed out, I can remove the `\w*` parts where it make sense to.